vaadin-button-flow-parent/vaadin-button-flow-integration-tests/src/test/java/com/vaadin/flow/component/button/tests/DetachReattachDisableOnClickButtonIT.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2000-2021 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.button.tests;
+
+import com.vaadin.flow.component.button.testbench.ButtonElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.tests.AbstractComponentIT;
+import org.junit.Assert;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+@TestPath("vaadin-button/detach-reattach-disable-on-click-button")
+public class DetachReattachDisableOnClickButtonIT extends AbstractComponentIT {
+
+    private ButtonElement getDisableOnClickButton() {
+        return $(ButtonElement.class).id("disable-on-click");
+    }
+
+    private void assertDisableOnClickButtonEnabled(
+            ButtonElement disableOnClickButton) {
+        Assert.assertTrue("'Disable on click' button should be enabled",
+                disableOnClickButton.isEnabled());
+    }
+
+    private void assertDisableOnClickButtonDisabled(
+            ButtonElement disableOnClickButton) {
+        Assert.assertFalse("'Disable on click' button should be disabled",
+                disableOnClickButton.isEnabled());
+    }
+
+    private void clickDisableOnClickButton(ButtonElement disableOnClickButton) {
+        getCommandExecutor().disableWaitForVaadin();
+
+        // Click 'Disable on click' button
+        disableOnClickButton.click();
+
+        // Check 'Disable on click' button is disabled
+        assertDisableOnClickButtonDisabled(disableOnClickButton);
+
+        waitUntil(ExpectedConditions.elementToBeClickable(
+                $(ButtonElement.class).id("disable-on-click")), 2000);
+
+        // Check 'Disable on click' button is enabled again
+        assertDisableOnClickButtonEnabled(disableOnClickButton);
+
+        getCommandExecutor().enableWaitForVaadin();
+    }
+
+    @Test
+    public void detachReattachDisableOnClickButtonTest() {
```

Comment:
Naming: It's not clear what this actually tests. The name should indicate that we want to assert that the button still gets disabled after detaching and reattaching. For example `testShouldBeDisabledOnClickAfterDetachingAndReattaching`

Reply:
Test was renamed.